src/activities/associateGrade/AssociateGradeEntity.js

```diff
@@ -1,22 +1,66 @@
 import { Entity } from '../../es6/Entity';
-import { Actions } from '../../hypermedia-constants';
+import { Actions, Classes } from '../../hypermedia-constants';
 import { performSirenAction } from '../../es6/SirenAction.js';
 
 /**
  * AssociateGrade entity of an activity.
  */
+const GRADEBOOK_STATUS = 'gradebookStatus';
+const GRADE_NAME = 'gradeName';
+const MAX_POINTS = 'maxPoints';
+
 export class AssociateGradeEntity extends Entity {
+
 	canEditGradebookStatus() {
 		return this._entity && this._entity.hasActionByName(Actions.activities.associateGrade.gradebookStatus);
 	}
 
-	async setGradebookStatus(newStatus) {
+	canEditNewGrade() {
+		const newGradeEntity = this._getNewGradeEntity();
+		return typeof newGradeEntity !== 'undefined' && newGradeEntity.hasActionByName(Actions.activities.associateGrade.chooseType);
```

Comment:
can just be `return newGradeEntity&& newGradeEntity.hasActionByName(Actions.activities.associateGrade.chooseType);`?

Reply:
Nope because it has to return `true` or `false`, not `undefined`. I had what you suggest initially, which failed tests that expect `false`.